design/slot-reservations-simplified.md

```diff
@@ -0,0 +1,282 @@
+request expiry.
+
+### Expanding window mechanism
+
+The expanding window mechanism prevents node and network overload once a slot
+becomes available to be filled (or repaired) by allowing only a very small
+number of SP addresses to fill/repair the slot at the start. Over time, the
+number of eligible SP addresses increases, until eventually all SP addresses in
+the network are eligible.
+
+The [expanding window
+mechanism](https://github.com/status-im/codex-research/blob/ad41558900ff8be91811aa5de355148d8d78404f/design/marketplace.md)
+starts off with a random source address, defined as $hash(block number, request
+id, slot index)$ and a distance defined as $XOR(A, A_0)$. Over time, $t_i$, the
+allowed distance [can be defined as $2^{256} *
+F(t_i)$](https://hackmd.io/@bkomuves/BkDXRJ-fC). As this value gradually
+increases, only addresses that have less of a distance than this value will be
+eligible to participate. In total, eligible addresses are those that satisfy:
+
+$XOR(A, A_0) < 2^{256} * F(t_i)$
+
+Because the source address for the expanding window is generated using the slot
+number, that means the source address for each slot will be different. Note that
+the reservation index is not included, meaning that a single node could
+potentially fill all slots in a request. The reason this was done was to
+simplify the expanding window design. The reservation index could be added in if
+necessary.
+
+The client can set the rate of expansion by defining the [parameter
+$h$](https://hackmd.io/@bkomuves/BkDXRJ-fC#Parametrizing-the-speed-of-expansion).
+Changing the value of $h$ will [affect the curve of the rate of
+expansion](https://www.desmos.com/calculator/pjas1m1472) (interactive graph).
+
+### Fill reward
+
+A fill reward will be issued to the SP that fills the slot. The client will deposit
+an additional fee when creating a request for storage to cover the maximum fill
+reward for all slots. Any difference in fill reward paid versus fill reward
+deposited will be returned to the client after the request is completed
+(including failed and cancelled requests).
+
+This reward will decrease linearly over time, starting with the maximum value
+at the time the slot is available to fill, and decreasing to zero at the request
+expiry. This incentivizes SPs to fill the slot as fast as possible, with
+the lucky few SPs that closest to the source point of the expanding window
+getting a bigger reward.
+
+The fill reward maximum value is specified by the client in the request for
+storage.
+
+#### Fill reward versus request collateral
+
+There is one caveat to the fill reward: if the fill reward is larger than the
+required collateral in an active request, an SP that is actively filling a slot
+will see a more profitable opportunity with a high fill reward (assuming SP's
+address was close to the source of the expanding window), and would be
+incentivized to abandon their existing duties and fill the slot in the new
+request.
```

Comment:
I doubt that this is an issue. I think Mark mentioned in one of our discussions that the fill reward is "secondary" here. The main goal for SPs is the request's reward... Also, the fill reward should be "small enough". Honestly, can't imagine that somebody would give out some large amount of money for it as the "main chunk" should go to reward.

Honestly, my question is should we set some lower bound of "minimum fill reward"? Should we support "zero fill reward"? I am bit thinking if we should not instead of leaving it up to the customer to set, to have some formula that would calculate it....